src/OrchardCore.Modules/OrchardCore.AdminDashboard/Drivers/DashboardContentDisplayDriver.cs

```diff
@@ -0,0 +1,109 @@
+            _contentManager = contentManager;
+
+        }
+
+        public override async Task<IDisplayResult> DisplayAsync(ContentItem model, BuildDisplayContext context)
+        {
+            var httpContext = _httpContextAccessor.HttpContext;
+            var dashboardFeature = httpContext.Features.Get<DashboardFeature>();
+
+            if (dashboardFeature == null || !dashboardFeature.IsManageRequest)
+            {
+                return null;
+            }
+
+            var results = new List<IDisplayResult>();
+
+            var hasPublished = await _contentManager.HasPublishedVersionAsync(model);
+            var hasDraft = model.HasDraft();
+            var hasEditPermission = await _authorizationService.AuthorizeAsync(httpContext.User, CommonPermissions.EditContent, model);
+            var hasDeletePermission = await _authorizationService.AuthorizeAsync(httpContext.User, CommonPermissions.DeleteContent, model);
+            var hasPublishPermission = await _authorizationService.AuthorizeAsync(httpContext.User, CommonPermissions.PublishContent, model);
+
+            var dragHandle = Dynamic("Dashboard_DragHandle", m =>
+            {
+                m.ContentItem = model;
+            }).Location("Leading:before");
+            results.Add(dragHandle);
+            if (hasEditPermission)
+            {
+                var editButton = Dynamic("Dashboard_EditButton", m =>
+                {
+                    m.ContentItem = model;
+                }).Location("ActionsMenu:after");
+                results.Add(editButton);
+
+            }
+
+            if (hasDeletePermission)
+            {
+                var deleteButton = Dynamic("Dashboard_DeleteButton", m =>
+                {
+                    m.ContentItem = model;
+                }).Location("ActionsMenu:after");
+                results.Add(deleteButton);
+            }
+
+
+            if (hasPublished && hasPublishPermission)
+            {
+                var unpublishButton = Dynamic("Dashboard_UnpublishButton", m =>
+                {
+                    m.ContentItem = model;
+                }).Location("ActionsMenu:after");
+                results.Add(unpublishButton);
+            }
+
+            if (hasDraft && hasPublishPermission)
+            {
+                var publishButton = Dynamic("Dashboard_PublishButton", m =>
+                {
+                    m.ContentItem = model;
+                }).Location("ActionsMenu:after");
+                results.Add(publishButton);
+
+            }
+
+            if (hasDraft && hasEditPermission)
+            {
+                var discardDraftButton = Dynamic("Dashboard_DiscardDraftButton", m =>
+                {
+                    m.ContentItem = model;
+                }).Location("ActionsMenu:after");
+                results.Add(discardDraftButton);
+            }
+            var shapeTag = Shape("DashboardWidget_DetailAdmin__ContentsTags", new ContentItemViewModel(model))
```

Comment:
Here to `Initialize<ContentItemViewModel>` is better

Reply:
done